lib/mutant/runner.rb

```diff
@@ -33,6 +33,12 @@ def initialize(*)
     def run_mutation_analysis
       @result = run_driver(Parallel.async(mutation_test_config))
       reporter.report(@result)
+
+      if env.config.json_dump
```

Comment:
I think instead of this branch, there should be a `Reporter::Sequence` that wraps any number of active reporters, calling `#start`, `#progress`, `#result` on each one. 

On the CLI you simply build up a `Reporter::Sequence` object by default containing an instance of `Reporter::CLI` but when the json dumping is active add an `Reporter::JSON` to the collection.

This would give you the same feature without infecting the runner on the special case.